dir/path.go

```diff
@@ -1,64 +1,104 @@
 package dir
 
 import (
+	"errors"
+	"io/fs"
+
 	"github.com/opencontainers/go-digest"
 )
 
 const (
+	// CertificateExtension defines the extension of the certificate files
+	CertificateExtension = ".crt"
+
+	// ConfigFile is the name of config file
+	ConfigFile = "config.json"
+
+	// KeyExtension defines the extension of the key files
+	KeyExtension = ".key"
+
+	// LocalKeysDir is the directory name for local key store
+	LocalKeysDir = "localkeys"
+
 	// SignatureExtension defines the extension of the signature files
 	SignatureExtension = ".sig"
+
+	// SignatureStoreDirName is the name of the signature store directory
+	SignatureStoreDirName = "signatures"
+
+	// SigningKeysFile is the file name of signing key info
+	SigningKeysFile = "signingkeys.json"
+
+	// TrustPolicyFile is the file name of trust policy info
+	TrustPolicyFile = "trustpolicy.json"
+
+	// TrustStoreDir is the directory name of trust store
+	TrustStoreDir = "truststore"
 )
 
 // PathManager contains the union directory file system and methods
 // to access paths of notation
 type PathManager struct {
-	ConfigFS  UnionDirFS
-	CacheFS   UnionDirFS
-	LibexecFS UnionDirFS
+	ConfigFS     UnionDirFS
+	CacheFS      UnionDirFS
+	LibexecFS    UnionDirFS
+	UserConfigFS UnionDirFS
+}
+
+func errorHandler(path string, err error) string {
+	// if path does not exist, return path for creating file.
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		panic(err)
+	}
+	return path
 }
 
 // Config returns the path of config.json
-func (p *PathManager) Config() (string, error) {
-	return p.ConfigFS.GetPath("config.json")
+func (p *PathManager) Config() string {
+	return errorHandler(p.ConfigFS.GetPath(ConfigFile))
 }
 
 // LocalKey returns path of the local private keys or certificate
 // in the localkeys directory
-func (p *PathManager) Localkey(keyName string) (string, error) {
-	return p.ConfigFS.GetPath("localkeys", keyName)
+func (p *PathManager) Localkey(keyName string) string {
```

Comment:
As mentioned in another pr, we can have 2 params, `keyName` and `extention` to enforce clients pass in valid filename.

Reply:
code updated.